app/invocation/invocation_exec_log_card.tsx

```diff
@@ -0,0 +1,321 @@
+        if (+(first?.spawn?.metrics?.totalTime?.seconds || 0) == +(second?.spawn?.metrics?.totalTime?.seconds || 0)) {
+          return +(first?.spawn?.metrics?.totalTime?.nanos || 0) - +(second?.spawn?.metrics?.totalTime?.nanos || 0);
+        }
+        return +(first?.spawn?.metrics?.totalTime?.seconds || 0) - +(second?.spawn?.metrics?.totalTime?.seconds || 0);
+    }
+    return 0;
+  }
+
+  handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const target = event.target;
+    const name = target.name;
+    this.setState({
+      [name]: target.value,
+    } as Record<keyof State, any>);
+  }
+
+  handleSortChange(event: React.ChangeEvent<HTMLSelectElement>) {
+    this.setState({
+      sort: event.target.value,
+    });
+  }
+
+  handleMnemonicFilterChange(event: React.ChangeEvent<HTMLSelectElement>) {
+    this.setState({
+      mnemonicFilter: event.target.value,
+    });
+  }
+
+  handleRunnerFilterChange(event: React.ChangeEvent<HTMLSelectElement>) {
+    this.setState({
+      runnerFilter: event.target.value,
+    });
+  }
+
+  handleMoreClicked() {
+    this.setState({ limit: this.state.limit + 100 });
+  }
+
+  handleAllClicked() {
+    this.setState({ limit: Number.MAX_SAFE_INTEGER });
+  }
+
+  getActionPageLink(entry: tools.protos.ExecLogEntry) {
+    const search = new URLSearchParams();
+    if (entry.spawn?.digest) {
+      search.set("actionDigest", digestToString(entry.spawn.digest));
+    }
+
+    return `/invocation/${this.props.model.getInvocationId()}?${search}#action`;
+  }
+
+  render() {
+    if (this.state.loading) {
+      return <div className="loading" />;
+    }
+    if (!this.state.log?.length) {
+      return (
+        <div className="invocation-execution-empty-state">
+          No execution log actions for this invocation{this.props.inProgress && <span> yet</span>}.
+        </div>
+      );
+    }
+
+    const mnemonics = new Set<string>();
+    const runners = new Set<string>();
+
+    const spawns = this.state.log
+      .filter((l) => {
+        if (l.spawn?.mnemonic) {
+          mnemonics.add(l.spawn.mnemonic);
+        }
+        if (l.spawn?.runner) {
+          runners.add(l.spawn.runner);
+        }
+        if (l.type != "spawn") {
+          return false;
+        }
+        if (this.state.mnemonicFilter != "" && l.spawn?.mnemonic != this.state.mnemonicFilter) {
+          return false;
+        }
+        if (this.state.runnerFilter != "" && l.spawn?.runner != this.state.runnerFilter) {
+          return false;
+        }
+        if (
+          this.props.filter != "" &&
+          !l.spawn?.targetLabel.toLowerCase().includes(this.props.filter.toLowerCase()) &&
+          !l.spawn?.mnemonic.toLowerCase().includes(this.props.filter.toLowerCase()) &&
+          !l.spawn?.args.join(" ").toLowerCase().includes(this.props.filter.toLowerCase()) &&
+          !l.spawn?.digest?.hash.toLowerCase().includes(this.props.filter.toLowerCase())
+        ) {
+          return;
```

Comment:
surprised this didn't whine at you, but i guess this should be `return false`, too

Reply:
Good catch, done!